diff --git a/base-notebook/start.sh b/base-notebook/start.sh
index 4fd80ae..3616014 100755
--- a/base-notebook/start.sh
+++ b/base-notebook/start.sh
@@ -59,6 +59,8 @@ if [ $(id -u) == 0 ] ; then
         done
     fi
 
+    HOMEDIR=/home/jovyan
+
     # handle home and working directory if the username changed
     if [[ "$NB_USER" != "jovyan" ]]; then
         # changing username, make sure homedir exists
@@ -69,16 +71,20 @@ if [ $(id -u) == 0 ] ; then
         fi
         # if workdir is in /home/jovyan, cd to /home/$NB_USER
         if [[ "$PWD/" == "/home/jovyan/"* ]]; then
-            newcwd="/home/$NB_USER/${PWD:13}"
-            echo "Setting CWD to $newcwd"
-            cd "$newcwd"
+            HOMEDIR="/home/$NB_USER/${PWD:13}"
+            echo "Setting CWD to $HOMEDIR"
+            # cd "$HOMEDIR"
         fi
     fi
 
     # Change UID of NB_USER to NB_UID if it does not match
     if [ "$NB_UID" != $(id -u $NB_USER) ] ; then
         echo "Set $NB_USER UID to: $NB_UID"
+        mkdir /tmp/homedir_shenanigans
+        usermod -d /tmp/homedir_shenanigans "$NB_USER"
         usermod -u $NB_UID $NB_USER
+        usermod -d "/home/$NB_USER" "$NB_USER"
+        rmdir /tmp/homedir_shenanigans
     fi
 
     # Set NB_USER primary gid to NB_GID (after making the group).  Set
@@ -86,7 +92,11 @@ if [ $(id -u) == 0 ] ; then
     if [ "$NB_GID" != $(id -g $NB_USER) ] ; then
         echo "Add $NB_USER to group: $NB_GID"
         groupadd -g $NB_GID -o ${NB_GROUP:-${NB_USER}}
+        mkdir /tmp/homedir_shenanigans
+        usermod -d /tmp/homedir_shenanigans "$NB_USER"
         usermod -g $NB_GID -a -G $NB_GID,100 $NB_USER
+        usermod -d "/home/$NB_USER" "$NB_USER"
+        rmdir /tmp/homedir_shenanigans
     fi
 
     # Enable sudo if requested
@@ -102,7 +112,7 @@ if [ $(id -u) == 0 ] ; then
     # the environment preserved
     run-hooks /usr/local/bin/before-notebook.d
     echo "Executing the command: $cmd"
-    exec sudo -E -H -u $NB_USER PATH=$PATH XDG_CACHE_HOME=/home/$NB_USER/.cache PYTHONPATH=$PYTHONPATH $cmd
+    exec sudo -E -H -u $NB_USER PATH=$PATH XDG_CACHE_HOME=/home/$NB_USER/.cache PYTHONPATH=$PYTHONPATH /bin/sh -c "cd \"$HOMEDIR\"; $cmd"
 else
     if [[ "$NB_UID" == "$(id -u jovyan)" && "$NB_GID" == "$(id -g jovyan)" ]]; then
         # User is not attempting to override user/group via environment
