diff --git a/jupyterhub/singleuser.py b/jupyterhub/singleuser.py
index 9a10834..43ceea4 100755
--- a/jupyterhub/singleuser.py
+++ b/jupyterhub/singleuser.py
@@ -472,7 +472,7 @@ class SingleUserNotebookApp(NotebookApp):
     def patch_templates(self):
         """Patch page templates to add Hub-related buttons"""
 
-        self.jinja_template_vars['logo_url'] = self.hub_host + url_path_join(self.hub_prefix, 'logo')
+        self.jinja_template_vars['logo_url'] = self.hub_host + url_path_join('static', 'base', 'images', 'IMAGE_HASH')
         self.jinja_template_vars['hub_host'] = self.hub_host
         self.jinja_template_vars['hub_prefix'] = self.hub_prefix
         env = self.web_app.settings['jinja2_env']
